packages/theming-core/src/generics/themeRegistry.ts

```diff
@@ -0,0 +1,247 @@
+/**
+ * Function which takes a parent theme, does some processing on it, and returns a new theme definition.
+ * This definition will then be passed to the standard resolver
+ *
+ * @internal This is an experimental interface and will be changed post design review.
+ */
+export type ProcessTheme<ITheme, IThemeDefinition> = (parentTheme: ITheme) => IThemeDefinition;
```

Comment:
minor note, because interfaces and types are both non-code-generating types that can often be interchanged, we have used I prefix. I don't know if there is a better way to identify, simply that we have that convention currently.

Reply:
(Note: a Class is TitleCased and it's obvious that it will generate code in your bundle. Consts do, Enums do as well. But types and interfaces do not generate code.)

Reply:
Tied a bit to my commend on `ResolveTheme`, what are the particular differences between `Process` and `Resolved`? What happens if no definition is passed to `ResolveTheme`, and how does that differ from `ProcessTheme`?